pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -233,7 +233,18 @@ func buildServices(hostAddresses []*HostAddress, name, namespace string, ports m
 				LabelSelectors:  selectors,
 			},
 			ServiceAccounts: saccounts,
-		})
+		}
+		if features.EnableDualStack && ha.address == constants.UnspecifiedIP {
+			// This is necessary so that dualstack wildcards are propagated to the listener
+			// for TCP listeners (e.g. HTTPS ServiceEntry ports)
+			s.ClusterVIPs = model.AddressMap{
```

Comment:
This feels a bit odd. Like ClusterVIPs should either always be set or never, not just when we have DS?

Maybe thats the only case we use it for today but that seems really coupled

Reply:
I was curious about that myself, but I wanted to make the change as narrow as possible. Will fix 👍🏾